notifications_utils/template.py

```diff
@@ -253,8 +254,35 @@ def fragment_count(self):
         Since we are supporting more or less "all" languages, it doesn't seem like we really want to count chars,
         and that counting bytes should suffice.
         """
-        content_len = len(self.content_with_placeholders_filled_in.encode("utf8"))
-        return math.ceil(content_len / 140)
+
+        # check if all chars are in the GSM-7 character set
+        def gsm_check(x):
+            rule = re.compile(r'^[\sa-zA-Z0-9_@?£!1$"¥#è?¤é%ù&ì\\ò(Ç)*:Ø+;ÄäøÆ,<LÖlöæ\-=ÑñÅß.>ÜüåÉ/§à¡¿\']+$')
+            if not rule.search(x):
+                return False
+            else:
+                return True
```

Comment:
Let's change this slightly just to make sure we're checking for the right thing.  According to [the Python re module documentation, `search()`](https://docs.python.org/3.9/library/re.html#re.search) returns `None` if no matches are found, which is a distinction between matching a pattern that just happens to be zero length.

In other words, the `None` return value is what we really want to test for to make sure the pattern wasn't found. 🙂 

We can also just test for the `None` return value and default to returning `True` to help simplify things a bit!

```suggestion
            gsm_match = rule.search(x)
            if gsm_match is None:
                return False
            return True
```